### PR TITLE
Split responses by correlation id more efficiently

### PR DESCRIPTION
**Pull request by rkatsyuryna**

### CODE REVIEW

File: core/src/main/java/io/github/tcdl/ChannelManager.java [hunk 1]
```diff
@@ -52,15 +52,12 @@ public Producer findOrCreateProducer(final String topic) {
...
-    public synchronized void subscribe(final String topic, final Consumer.Subscriber subscriber) {
-        Consumer consumer = findOrCreateConsumer(topic);
-        consumer.subscribe(subscriber);
-    }
-
-    private Consumer findOrCreateConsumer(final String topic) {
+    public Consumer subscribe(String topic, Subscriber subscriber) {
         Validate.notNull(topic, "field 'topic' is null");
+        Validate.notNull(subscriber, "field 'subscriber' is null");
         Consumer consumer = consumersByTopic.computeIfAbsent(topic, key -> {
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/ChannelManager.java:57]: What if consumer already exist? Maybe we should explicitly throw an error

File: core/src/main/java/io/github/tcdl/Collector.java [hunk 2]
```diff
@@ -96,13 +97,13 @@ public boolean isAwaitingResponses() {
     public void listenForResponses(String topic, Message requestMessage) {
         this.topic = topic;
         this.requestMessage = requestMessage;
-        channelManager.subscribe(this.topic, this);
+        collectorSubscriber.registerCollector(topic, this);
+        channelManager.subscribe(this.topic, collectorSubscriber);
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/Collector.java]: looks like a hack. We overwrite the subscribe over and over again

File: core/src/main/java/io/github/tcdl/ChannelManager.java (hunk 1 above)
Comment (rkatsyuryna) [inline core/src/main/java/io/github/tcdl/ChannelManager.java:57]: not this is ok, e.g we have few collectors that try to concurrently consume from topic. First one will win and creaae one. Others just need this ONE. Only interest is you get this ONE

File: core/src/main/java/io/github/tcdl/Collector.java (hunk 2 above)
Comment (rkatsyuryna) [inline core/src/main/java/io/github/tcdl/Collector.java]: no this only the name, actually subscribe is allias for "old" findOrCreateStuff - first time we create Consumer we do subscribe. All next calls will cause nothing

**Pull request merged**